common/djangoapps/entitlements/views.py

```diff
@@ -0,0 +1,242 @@
+import uuid
+from datetime import datetime
+
+from django.contrib.auth.models import User
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions, status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from student.models import CourseEnrollment
+from .models import CourseEntitlement
+
+SESSION_CLASSES = (JwtAuthentication, SessionAuthentication)
+PERMISSION_CLASSES = (permissions.IsAuthenticated, permissions.IsAdminUser)
+
+
+def _get_updated_entitlement_data_fields(request):
+    """
+    Get the updated entitlement data fields from the request
+
+    Arguments:
+        request: Request object
+    Returns:
+        Dict: A dictionary containing only the updated data fields
+    """
+    request_data = {
+        'course_uuid': request.data.get('course_uuid', None),
+        'mode': request.data.get('mode', None),
+        'username': request.data.get('username', None),
+        'order_number': request.data.get('order_number', None),
+    }
+    return dict((k, v) for k, v in request_data.iteritems() if v)
+
+
+def _get_response_object(course_entitlements):
+    """
+    Build the response dictionary for the Response
+
+    Arguments:
+        course_entitlements (list): A list containing 1 or more CourseEntitlements
+    Returns:
+        Dict: A dictionary containing the results list
+    """
+    results_list = []
+    for entitlement in course_entitlements:
+        results_list.append(entitlement.to_json())
+    return {'results': results_list}
+
+
+class EntitlementView(APIView):
+    authentication_classes = SESSION_CLASSES
+    permission_classes = PERMISSION_CLASSES
+
+    def _get_clear_entitlements_params_filter_data(self, request):
+        """
+        Get all the parameter Keys and values and clear out any that are None/not available
+
+        Arguments:
+            request: Request Object
+        Returns:
+            Dict: Dictionary containing all the Parameters that are not None
+        """
+        request_data = {
+            'uuid': request.query_params.get('uuid', None),
+            'mode': request.query_params.get('mode', None),
+            'username': request.query_params.get('username', None),
+            'order_number': request.query_params.get('order_number', None),
+        }
+        return dict((k, v) for k, v in request_data.iteritems() if v)
+
+    def get(self, request):
```

Comment:
You could probably do a lot less work if you use DRF a bit more

implement  get_queryset and list.  And use a filter backend to filter on the entitlement model fields.